equinix/helpers_device.go

```diff
@@ -271,3 +319,42 @@ func ipAddressSchema() *schema.Resource {
 		},
 	}
 }
+
+func getDeviceMap(device metalv1.Device) map[string]interface{} {
```

Comment:
this function can be used in the singular datasource, and maybe also in device resources